server/data/data-jmap/src/test/java/org/apache/james/jmap/api/filtering/FilteringManagementContract.java

```diff
@@ -66,7 +66,7 @@ default void listingRulesShouldThrowWhenNullUser(EventStore eventStore) {
     default void listingRulesShouldReturnDefinedRules(EventStore eventStore) {
         FilteringManagement testee = instantiateFilteringManagement(eventStore);
 
-        Mono.from(testee.defineRulesForUser(USERNAME, RULE_1, RULE_2)).block();
+        Mono.from(testee.defineRulesForUser(USERNAME, Optional.of(Version.INITIAL), RULE_1, RULE_2)).block();
```

Comment:
Just pass `Optional.empty()` for all existing tests!

Reply:
You mean when the first time set filter, the state/version should be empty, right ? I got it.
But when we do Filter/get for user dont have filter yet, it return Version.INITIAL(-1).
Is that a kind of lack of consistency? Shouldn't it return Optional.empty ?

Reply:
```
    public Rules listRules() {
        return new Rules(state.rules,
            history.getVersionAsJava().map(EventId::value).map(Version::new).orElse(Version.INITIAL));
    }
```